Source/VisualRecognitionV3/VisualRecognition+CoreML.swift

```diff
@@ -208,17 +207,20 @@ extension VisualRecognition {
                     let requestHandler = VNImageRequestHandler(data: imageData)
                     try requestHandler.perform([request])
                 } catch {
-                    dispatchGroup.leave()
                     let error = WatsonError.other(message: "Failed to process classification request: \(error.localizedDescription)")
-                    completionHandler(nil, error)
+                    errors.append(error)
+                    dispatchGroup.leave()
                     return
                 }
             }
         }
 
         // return results after all classification requests have executed
         dispatchGroup.notify(queue: DispatchQueue.global(qos: .userInitiated)) {
-            guard !results.isEmpty else { return }
+            guard errors.isEmpty else {
+                completionHandler(nil, errors[0])
+                return
```

Comment:
True. How would you propose that we deliver _both_ successful results _and_ errors with a single completion handler?

Reply:
Would it be reasonable to hit the completion handler with non-nil values for both the result and error?

Reply:
I just pushed a change to return any successful results along with an error to indicate that some local classifications failed.

I'm still only returning information about the first error, because the `.other` error enum does not currently have a `metadata` associated value to store information about the additional errors (if any). I've created a PR to add `metadata` to the `.other` error enum in the next major release, at which point we can update this code to return the full set of errors.